File: src/types/workflow.ts
import { BaseEntity } from './common';

// Workflow types
export interface WorkflowComponent {
  id: string;
  label: string;
  type: string;
  link?: string | null;
  position: {
    x: number;
    y: number;
  };
  config?: Record<string, any>;
}

export interface WorkflowEdge {
  source: string;
  target: string;
}

export interface Workflow extends BaseEntity {
  name?: string;
  description?: string;
  agent_id: string;
  nodes: WorkflowComponent[];
  edges: WorkflowEdge[];
  status?: 'draft' | 'active' | 'inactive';
}

export interface CreateWorkflowRequest {
  name: string;
  description?: string;
  agent_id: string;
  nodes?: WorkflowComponent[];
  edges?: WorkflowEdge[];
}

export interface UpdateWorkflowRequest extends Partial<CreateWorkflowRequest> {}

export interface WorkflowComponentType {
  id: string;
  name: string;
  description: string;
  category: string;
  properties: Record<string, any>;
}

export interface WorkflowExecution {
  id: string;
  workflow_id: string;
  status: 'running' | 'completed' | 'failed' | 'cancelled';
  started_at: string;
  completed_at?: string;
  error_message?: string;
  execution_data?: Record<string, any>;
}

File: agent-api-server/app/schemas/workflow.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class WorkflowResponse(BaseModel):
    id: str
    name: str
    description: str
    nodes: List[Dict[str, Any]]
    edges: List[Dict[str, Any]]
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class WorkflowCreate(BaseModel):
    name: str
    description: Optional[str] = None
    nodes: List[Dict[str, Any]]
    edges: List[Dict[str, Any]]
    status: str = "active"


class WorkflowUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    nodes: Optional[List[Dict[str, Any]]] = None
    edges: Optional[List[Dict[str, Any]]] = None
    status: Optional[str] = None
